libr/util/list.c

```diff
@@ -116,20 +120,30 @@ R_API void r_list_split (RList *list, void *ptr) {
 	}
 }
 
-R_API void r_list_split_iter (RList *list, RListIter *iter) {
-	if (list->head == iter) list->head = iter->n;
-	if (list->tail == iter) list->tail = iter->p;
-	if (iter->p) iter->p->n = iter->n;
-	if (iter->n) iter->n->p = iter->p;
+R_API void r_list_split_iter(RList *list, RListIter *iter) {
```

Comment:
I think this function should decrement `list->length` because it removes an element from the list.